src/common/BaseComponent.ts

```diff
@@ -47,6 +52,17 @@ export class BaseComponent<P, S> extends React.Component<P, S> {
     return (results && results.length > 1) ? results[1] : '';
   }
 
+  /**
+   * Gives the class constructor, will iterate through prototype methods prefixed with "_on" and bind them to "this".
+   * Example: in your constructor, you'd have: this.autoBindCallbacks(MyComponent); */
+  protected autoBindCallbacks(object: Function) {
```

Comment:
I like that suggestion. I'll remove it here and approach it that way separately.